sshnodelay.c

```diff
@@ -5,6 +5,32 @@
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 
+#if __APPLE__
+
+int custom_connect(int sock, const struct sockaddr *addr, socklen_t addrlen);
+
+typedef struct interpose_s {
+	void *new_func;
+	void *orig_func;
+} interpose_t;
+
+static const interpose_t interposers[] \
+	__attribute__ ((section("__DATA, __interpose"))) = {
+	{ (void *)custom_connect,  (void *)connect  },
+};
```

Comment:
Sure. I was going to say "don't know", but then I had to look around and found: http://toves.freeshell.org/interpose/interpose_old.html . Pretty interesting.

I have added a comment and taken the chance to remove code duplication in this file.

(Haven't build-tested this specific change on Linux yet; will have to do so tomorrow from work.)